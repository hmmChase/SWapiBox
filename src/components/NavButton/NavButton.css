button {
  font-family: 'Exo 2', sans-serif;
  font-weight: 600;
  background: none;
  border: 2px solid seagreen;
  cursor: pointer;
  font-size: 20px;
  padding: 10px;

}

button:hover {
  background:rgba(0,0,0,0.1);
}