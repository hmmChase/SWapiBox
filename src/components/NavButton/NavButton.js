import React from 'react';
import PropTypes from 'prop-types';
import './NavButton.css';

const NavButton = (props) => {

  return (
    <button
      onClick={() => props.setCategory(props.category)}
    >{props.category}
    </button>
  );
};

NavButton.propTypes = {
  category: PropTypes.string,
  setCategory: PropTypes.func
};


export default NavButton;
