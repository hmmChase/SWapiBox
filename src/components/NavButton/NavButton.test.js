import React from 'react';
// import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import NavButton from './NavButton';

let mockProps = {
  category: 'mock category',
  setCategory: jest.fn()
};

describe('NavButton', () => {
  let navButton;

  beforeEach(() => {
    navButton = shallow(<NavButton {...mockProps} />);
  });

  it('matches the snapshot', () => {
    expect(navButton).toMatchSnapshot();
  });

  it('calls props.setCategory when clicked with correct args', () => {
    const button = navButton.find('button');
    button.simulate('click');

    expect(mockProps.setCategory).toHaveBeenCalledTimes(1);
    expect(mockProps.setCategory).toHaveBeenCalledWith('mock category');
  });
});
