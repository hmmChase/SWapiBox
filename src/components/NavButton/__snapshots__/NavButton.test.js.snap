// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavButton matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NavButton
    category="mock category"
    setCategory={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "mock category",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": "mock category",
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "mock category",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": "mock category",
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
